### PR TITLE
Fix examples to use new config format from TGen v1.0.0

### PR DESCRIPTION
**Pull request by robgjansen**
Closes GH-100

### CODE REVIEW

**Comment by robgjansen** [comment]
> Looks like this breaks the CI.

Yeah, sorry, it wasn't ready for review yet. I suppose I should stop opening pull requests until I'm actually ready :S

**Comment by sporksmith** [comment]
Odd; in the history there's a "request review" event at the same time that the PR was created. Maybe Github "helpfully" auto-adding a reviewer?

**Comment by robgjansen** [comment]
Oh I bet I checked your name when I created the pull request, but then after it was created the CI got triggered and then I noticed the CI failure but forgot to uncheck your name. Let's call this my bad, and lesson learned :)

**Comment by robgjansen** [comment]
> Btw any thoughts about not compressing these files, so that git can show diffs etc? I know they're a bit large, but IIUC the historical versions are internally compressed anyway.

Some of the shadow.data.template files need to have certain permissions preserved in order for the experiment to complete successfully (Tor requires certain permissions are set on keys etc.), and it turns out git doesn't preserve these permissions by default. The files will be created with the mask of the user that is doing the checkout. Using tarballs means we do preserve the permissions. I think committing the decompressed versions will lead to many people failing to run the examples correctly.

Unless I'm missing something and there is a way to force git to preserve file permissions?

**Comment by sporksmith** [comment]
Ah, that's unfortunate. I suppose there could be alternatives like having a script that sets the permissions properly, but I'm fine with sticking with the tarball for now.

**Comment by robgjansen** [comment]
I can't get Tor to function correctly. Occasionally, the first stream request from TGen to the Tor SOCKS proxy fails, with Tor returning an error code that indicates a "general error". But after that all SOCKS requests succeed. I haven't been able to figure out why because Tor's info logs are not helpful. I haven't explored logging at debug level yet, but I've already spent considerable time on this PR. I need to set it aside for a while.

**Comment by robgjansen** [comment]
Noting this option for the future, we could write a script to fix the permissions and install it in a git hook that runs after a pull.
https://git-scm.com/docs/githooks

This may have been what @sporksmith suggested too :)

**Comment by GitZinger** [comment]
Hi guys,
Can I ask do the sample examples work? Can I git update or something to try it along with shadow, TGen and Tor-plug-in?

**Comment by robgjansen** [comment]
It should mostly work. You can try by cloning with
```
git clone git@github.com:robgjansen/shadow-plugin-tor.git -b example-fixup
```
Then see the example in `shadow-plugin-tor/resource/shadowtor-minimal-config.tar.xz`

**Comment by GitZinger** [comment]
> It should mostly work. You can try by cloning with
> 
> ```
> git clone git@github.com:robgjansen/shadow-plugin-tor.git -b example-fixup
> ```
> 
> Then see the example in `shadow-plugin-tor/resource/shadowtor-minimal-config.tar.xz`

Permission Denied(publickey)
fatal: could not read from remote repository.

Could you check the permission for me please?
thank you

**Comment by robgjansen** [comment]
Try this:

```
git clone https://github.com/robgjansen/shadow-plugin-tor.git -b example-fixup

```

**Comment by GitZinger** [comment]
yeah, it is working. thank you guys.
but another question:
https://github.com/shadow/shadow-plugin-tor/wiki#Running%20Tor%20in%20Shadow
the run command is 
`shadow shadow.config.xml > shadow.log`
afterwards, the information goes into **shadow.log**
but in the following instructions:
`for d in shadow.data/hosts/*client*; do grep "stream-success" ${d}/* ; done | tee clients.log | wc -l`
the **clients.log** should be replaced with **shadow.log**  or something?
I am new to here! I really appreciate your advise!

**Comment by robgjansen** [comment]
> but another question:

@GitZinger This is a pull request and the discussion should stay focused around it. If you want to ask a more general question, please open a new issue for that so that we can keep our development tasks organized. Thank you.

**Comment by robgjansen** [comment]
The error from the CI is: "Expected 40 successful streams got 37".

So there seems to be a bug in the SOCKS handshake between TGen and Tor (for the initial handshake only, as described above). My guess is that the bug was introduced somewhere between TGen v0.0.1 and TGen v1.0.0. I don't think blocking this configuration update due to a bug in a version of TGen that people are using is a good idea. As it stands, the configs in master are known to be broken with the latest version of TGen, and several users have raised bugs due to the old TGen configs. This PR updates the old configs, even if it doesn't fix every other bug in TGen v1.0.0.

So, @sporksmith, I propose that we merge this PR even though the CI is failing.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -15,20 +15,25 @@ on:
...
+    runs-on: ubuntu-18.04
     strategy:
       matrix:
         # TODO: add clang. Currently builds but has unsuccessful xfers in simulation.
         cc: ['gcc']
         # Latest patch version from last few release branches.
         # See https://gitweb.torproject.org/tor.git/refs/tags
-        tor_version: ['0.3.5.7', '0.4.0.5', '0.4.1.8', '0.4.2.6']
+        # TODO add more recent versions of Tor
+        # FIXME Tor v0.4.2.6 was previously disabled since it failed the CI
```
Comment (sporksmith) [inline .github/workflows/build_and_test.yml:26]: Maybe open an issue and add a reference to it here?

**Review by sporksmith — APPROVED** [review]

**Pull request merged**